### PR TITLE
smoothcontrol: Don't process throttle signal when it's set to zero.

### PR DESCRIPTION
**Pull request by mlyle**
This is just the safety fix from #1950 to merge sooner/speedily.

### CODE REVIEW

File: flight/Libraries/math/smoothcontrol.c [hunk 1]
```diff
@@ -117,9 +117,11 @@ void smoothcontrol_run(smoothcontrol_state state, uint8_t axis_num, float *new_s
...
 void smoothcontrol_run_thrust(smoothcontrol_state state, float *new_signal)
 {
-	// The -1 on zero throttle causes a huge blip when throttling up, so we need to
-	// handle it separately.
-	if(*new_signal < 0)
+	/* The -1 on zero throttle causes a huge blip when throttling up, so we need to
+	   handle it separately.
+	   Also, reset the state, if zero throttle is requested, so that none of the modes
+	   can keep it non-zero. */
+	if(*new_signal <= 0)
```
Comment (tracernz) [inline flight/Libraries/math/smoothcontrol.c]: Can you please add the missing space here while you're changing the line?

Comment (mlyle) [reply flight/Libraries/math/smoothcontrol.c]: OK, I fixed up his formatting in this neighborhood.

**Review by tracernz — APPROVED** [review]
Thanks.

**Comment by mlyle** [comment]
Thx, I feel a lot better with this fixed.

**Pull request merged**